includes/admin/reports/class-wc-admin-report.php

```diff
@@ -558,7 +558,7 @@ public function calculate_current_range( $current_range ) {
 			break;
 
 			case '7day' :
-				$this->start_date    = strtotime( '-6 days', current_time( 'timestamp' ) );
+				$this->start_date    = strtotime( '-6 days', strtotime( 'midnight', current_time( 'timestamp' ) ) );
```

Comment:
Actually, after doing some more research, I think the best way of doing this is `strtotime( 'today midnight - 6 days')`.